lib/logstash/util/java_version.rb

```diff
@@ -0,0 +1,66 @@
+require 'cabin'
+
+module LogStash::Util::JavaVersion
+  def self.logger
+    @logger ||= Cabin::Channel.get(LogStash)
+  end
+
+  # Print a warning if we're on a bad version of java
+  def self.warn_on_bad_java_version
+    if self.bad_java_version?(self.version)
+      msg = "!!! Please upgrade your java version, the current version '#{self.version}' may cause problems. We recommend a minimum version of 1.7.0_51"
+      STDERR.puts(msg)
+      logger.warn(msg)
+    end
+  end
+
+  # Check to see if this is a recommended java version, print a warning to stdout if this is a bad version
+  # Returns the current java version
+  def self.version
+    return nil unless LogStash::Environment.jruby?
+
+    require 'java'
+    java_import "java.lang.System"
```

Comment:
java_import isn't necessary, you can just do `java.lang.System.getProperty("java.runtime.version")` directly w/o java_import. 

I think `require "java"` is also not needed as some time ago JRuby requires that by default, I think? At least on JRuby 1.7.19 it seems to:

```
% rvm use 1.7.19
Using /Users/jls/.rvm/gems/jruby-1.7.19
% ruby -e 'puts require "java"'
false
```
(false means it was already required)